src/btree/bt_delete.c

```diff
@@ -167,6 +167,15 @@ __wt_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
         goto err;
     if (addr.ta.prepare)
         goto err;
+
+    /*
+     * When performing a truncate operation with no associated timestamp, limit fast-truncate to
+     * pages where all its data is globally visible.
+     */
+    if (F_ISSET(session->txn, WT_TXN_TS_NOT_SET) &&
+      !__wt_txn_visible_all(session, addr.ta.newest_txn, addr.ta.newest_start_durable_ts))
```

Comment:
Here we should use WT_MAX(addr.ta.newest_start_durable_ts, addr.ta.newest_stop_durable_ts) to decide the global visibility.